### ISSUE TITLE
Status updates for OTF staff

### ISSUE THREAD

**Issue by gibbsmaya** [OpenTechFund/opentech.fund#792]
**Acceptance criteria**

- [x] Updates to statuses for OTF staff as shown in #786 

**QA criteria**
Dev:

- [x] checked feature meets acceptance criteria/conforms exactly to the specification.
- [x] provided good unit test coverage (if this is non-trivial behaviour).
- [x] checked all tests for the project pass with this feature enabled.
- [x] checked code conforms to the project coding standards.
- [x] had code reviewed by another developer and resolved any issues raised.
- [x] tested this feature as an end user of the website/app (Can I get to it? Is it useable? Can I break it? Does it work in an end-to-end context?)
- [x] checked that this feature works on the server/s I am deploying it to.

QA:

 * [x] tested this feature as a front end user and it meets the acceptance criteria/conforms to the specification and design.
 * [x] checked that the feature works on the server/s deployed to

**Comment by todd-dembrey** [comment]
Blocked waiting on a discussion with @tmsndrs

**Comment by todd-dembrey** [comment]
This is deployed for reivew and being code reviewed by @emullaney in #838

**Comment by gibbsmaya** [comment]
Hi Todd,

I have started testing this and have come across the following issues before getting blocked:

 - First status in concept note should be screening.
 - Has ready for determination status been added? I can't see it in the status bar.
 - When progressing an application from internal review to ready for discussion (action: close review) I get a 500 error.

I am testing on a new submission (https://dev-apply.opentech.fund/apply/submissions/12/). I now seem to be able to access the submission as OTF staff and it has moved to ready for discussion status but as the applicant I am getting a 500 error when trying to access even the dashboard. Can you look into this before I continue testing?

Thanks,
Maya

**Comment by gibbsmaya** [comment]
When application is in ready for discussion, the application displays in 'ready for determination' list on dashboard for OTF staff. Is this intentional?

Once invited to proposal, I don't think we need the second status of 'ready for discussion'. I think this should go straight to OTF review. Or should this be 'proposal received? Check with Tom.

I would like us to show the full status bar on the submission page rather than just the concept note/proposal part, as per the dashboard status bar. I can split this into a separate ticket though if we want to tackle it separately. To discuss with Tom.

**Comment by todd-dembrey** [comment]
Following changes made and available on dev:

* Concept - First status changed to "Screening"
* Ready For Determination added as a new phase for all workflows. Can be transitioned to from any discussion/screening phase and any phase that currently has ability to accept/reject.

**Comment by todd-dembrey** [comment]
I have made the relevant changes and deployed.

To confirm, i have removed the ready for determination block from the dashboard as it is not in the new designs

**Comment by gibbsmaya** [comment]
I also think the ordering of statuses in the update status drop down from the action box on submission page should follow the ordering of the workflow - currently this is confusing.

**Comment by gibbsmaya** [comment]
Displaying full status bar for 2 phases does work for applicants. Remaining as single phase for OTF staff. Will split out my last comment about ordering of statuses in status updates box into a separate issue.

**Comment by gibbsmaya** [comment]
Marking as done but requires code review from @emullaney

**Comment by gibbsmaya** [comment]
@todd-dembrey please can you deploy this to test?

**Comment by gibbsmaya** [comment]
Hi @fourthletter, @danblah

This work is now ready for testing. This work consists of the new statuses for the workflow states an application goes through. This is best seen from the status bar on the application page:
![screen shot 2019-01-22 at 09 10 23](https://user-images.githubusercontent.com/19690895/51524258-94c3e480-1e25-11e9-91e0-cbdce58c25f0.png)
You will see that labels have been updated as well as some additional new states as discussed in the demo call. As a brief summary, the statuses mean the following:

**Concept note**

 - Screening: Initial phase in your workflow whereby the round lead reviews received submissions within the round and apply's an initial screening status, e.g. 'in remit', 'out of remit'.
 - Internal review: Phase for OTF to review the applications, normally two reviews.
 - Ready for discussion: When 2 reviews have been completed submissions are moved into this state and the applications in this state become your agenda for the review call.
 - Ready for preliminary determination: An optional status for when the preliminary determination is known, but needs further time to action. This state allows OTF staff to move it out of the review call agenda and create a list for round leads to action final determinations on the concept note. e.g. you decide in the review call that you are going to invite this concept not to proposal. You move it into ready for preliminary determination so it is clear it is actioned and no longer needs discussion, however it can't move into preliminary determination state because it requires an action for the round lead to write a bespoke invitation for proposal to the applicant.
 - Preliminary determination: Decision on the concept note - either dismissed or invite to proposal.

**Proposal**

 - Invited for proposal: Applicant has been invited for proposal bit has not yet submitted proposal.
 - Proposal received: Applicant has submitted proposal.
 - Internal review: Phase for OTF to review the applications, normally two reviews.
 - Ready for discussion: When 2 reviews have been completed submissions are moved into this state and the applications in this state become your agenda for the review call.
 - Advisory council review: Phase for Advisory council to review the applications, normally two reviews.
 - Ready for discussion: When 2 reviews have been completed submissions are moved into this state and the applications in this state become your agenda for the review call.
 - Ready for final determination: An optional status for when the preliminary determination is known, but needs further time to action. This state allows OTF staff to move it out of the review call agenda and create a list for round leads to action final determinations on the concept note. e.g. you decide in the review call that you are going to accept this proposal. You move it into ready for final determination so it is clear it is actioned and no longer needs discussion, however it can't move into application outcome state because it requires an action for the round lead to write a bespoke determination message to the applicant.
 - Application outcome: Decision on the proposal - either dismissed or accepted.

Please can you test both types of application and ensure you are happy with the states? I suggest you move an application through the full workflow of both concept and proposal, using a test applicant account with your email address so you can also check what emails are coming through on status changes. Slack message got to `#webapp-test` so you can also review what slack messages are going out and check they are correct. Efficiency-wise it is worth you testing this ticket alongside the #818 , reviewing what the applicant sees for each OTF status update. Details of the applicant statuses are outlined in that issue.

Please test this against the acceptance criteria noted in the initial issue description.

Please let us know if anything is unclear or if you have any questions. Otherwise please provide feedback of any issues you find or if everything is working as expected, please confirm and we can get this work deployed to the live site and issue closed off.

Remember that it is really easy for us to change the labelling on the status names so let us know if any of these don't make sense.

Thanks,
Maya

**Comment by danblah** [comment]
Looks good, acceptance criteria met from my testing. What effects does this have on single stage applications, both w/external and w/o external review?

**Comment by gibbsmaya** [comment]
See my response here https://github.com/OpenTechFund/opentech.fund/issues/818#issuecomment-456420883 - same applies. Please do test this case and ensure you are happy with it.

**Comment by gibbsmaya** [comment]
@danblah @fourthletter can you confirm if you are happy for this to me moved into production deployment?

**Comment by fourthletter** [comment]
- All acceptance criteria in this issue appears to be accurate and functional. Thanks! 💥
- I don't have the admin authority to change a submission's email to receive status updates, but I can see these updates via #webapp-test

**Comment by gibbsmaya** [comment]
Thanks @fourthletter. Moving to @todd-dembrey for deployment to production.

**Comment by fourthletter** [comment]
All acceptance criteria are visible and functional during today's test. Thanks! 💥
